### ISSUE TITLE
Change the message displayed by NotAMockException to include the class name

### ISSUE THREAD

**Issue by GoogleCodeExporter** [dlsgusrn7577/mockito#224]
```
Change the message displayed by NotAMockException to include the class name of 
the object that is not a mock. 
This will easily avoid simple syntax mistakes such misplacing a parentheses: 
Mockito.verify( mockInstance.method() );
instead of
Mockito.verify( mockInstance ).method();
Where a Boolean is passed to the verify method, instead of the mockInstance, 
which isn't clear from the exception. 

The advantage is that, instead of having to debug the issue, it would be so 
much easier if the exception message just said: 
org.mockito.exceptions.misusing.NotAMockException:
Argument passed to verify() is of type java.lang.Boolean and is not a
mock!
instead of just
org.mockito.exceptions.misusing.NotAMockException:
Argument passed to verify() is not a mock!

This applies to the 1.8.5 version of Mockito. 
```

Original issue reported on code.google.com by `hanrisel...@gmail.com` on 15 Oct 2010 at 12:04

**Comment by GoogleCodeExporter** [comment]
```
Makes perfect sense.
```

Original comment by `szcze...@gmail.com` on 18 Oct 2010 at 12:37
- Added labels: **Priority-High**
- Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
Fixed by revision 966a8bc0e0
```

Original comment by `brice.du...@gmail.com` on 1 Nov 2010 at 10:50
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
Original comment by `szcze...@gmail.com` on 3 Jul 2011 at 12:43
- Added labels: **Milestone-Release1.9-rc1**
